planner/core/planbuilder.go

```diff
@@ -662,6 +662,21 @@ func (b *PlanBuilder) getPossibleAccessPaths(indexHints []*ast.IndexHint, tblInf
 	return available, nil
 }
 
+func (b *PlanBuilder) filterPathByIsolationRead(paths []*accessPath) ([]*accessPath, error) {
+	// TODO: filter paths with isolation read locations.
+	isolationReadEngines := b.ctx.GetSessionVars().GetIsolationReadEngines()
+	for i := len(paths) - 1; i >= 0; i-- {
+		if _, ok := isolationReadEngines[paths[i].storeType]; !ok {
+			paths = append(paths[:i], paths[i+1:]...)
+		}
+	}
+	var err error
+	if len(paths) == 0 {
+		err = ErrInternal.GenWithStackByArgs("No access path for table reader after isolation read filter")
```

Comment:
```suggestion
		err = ErrInternal.GenWithStackByArgs("Can not find access path matching \"%v\"(value: \"%v\"). Available values are \"%v\".", TiDBIsolationReadEngines, valueOf(TiDBIsolationReadEngines), availableValueOf(TiDBIsolationReadEngines))
```

Reply:
Why need to print `Available values` here? This is not the place to set values.